MixStaticSeq/seq_ehr_model.py

```diff
@@ -98,23 +123,40 @@ def forward(self, x=None):
         # non_seq_rep: (B, h)   seq_rep: (B, h)
         m_rep = torch.cat([non_seq_rep, seq_rep], dim=1)
 
-        # TODO we need to work on this part of the network: test different non-linear function; test number of layers
+        # merge (B, h+h)
         raw_rep = self.merge_layer(m_rep)
-        m_rep = torch.tanh(F.dropout(raw_rep, p=self.dropout_rate))
+        m_rep = F.dropout(raw_rep, p=self.dropout_rate)
 
         # (B, 2)
         logits = self.classifier(m_rep)
         pred_prob = F.softmax(logits, dim=-1)
 
         if self.loss_mode is ModelLossMode.BIN:
             # y dim (B, 2)
-            loss = F.binary_cross_entropy(pred_prob, y)
-            # loss = F.binary_cross_entropy_with_logits(logits, y)
+            loss = F.binary_cross_entropy_with_logits(logits, y, weight=self.sampling_weight)
         elif self.loss_mode is ModelLossMode.MUL:
             # y dim (B, 1)
-            # y_hat = y.type(torch.long)
-            loss = F.cross_entropy(logits, y)
+            loss = F.cross_entropy(logits, y, weight=self.sampling_weight)
         else:
             raise NotImplementedError("loss mode only support bin or mul but get {}".format(self.loss_mode.value))
 
         return loss, pred_prob, torch.argmax(pred_prob, dim=-1), m_rep
+
+
+class MixEmbeddingModel(MixModel):
```

Comment:
This MixEmbeddingModel class required extra flags for the embedding file and embedding options.  I am thinking to separate this and MixTCNModel into an individual module.